src/mount.rs

```diff
@@ -0,0 +1,655 @@
+use lru::LruCache;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use std::{
+    borrow::Cow,
+    cmp::Ordering,
+    ffi::OsStr,
+    fs,
+    io::{self, ErrorKind, Read},
+    iter::Peekable,
+    num::NonZeroUsize,
+    ops::ControlFlow,
+    path::{Component, Path, PathBuf},
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use tracing::{error, info, trace};
+
+use bytes::Bytes;
+use itertools::Itertools;
+use tracing::{debug, warn};
+use windows_projfs::{
+    DirectoryEntry, DirectoryInfo, FileInfo, Notification, ProjectedFileSystem,
+    ProjectedFileSystemSource,
+};
+
+use crate::{
+    counters::Counter,
+    monitor::{
+        task::{Task, TaskList},
+        Monitor, Problem,
+    },
+    Apath, Archive, BandId, BandSelectionPolicy, IndexEntry, IndexRead, Kind, Result, StoredTree,
+};
+
+macro_rules! static_dir {
```

Comment:
This is (currently) used only once so perhaps it's no worth having as a macro? It could just be a function, or even just inline?